front/inventory.php

```diff
@@ -46,7 +46,15 @@
     $refused->getFromDB($_GET['refused']);
     $contents = file_get_contents($refused->getInventoryFileName());
 } else if ($_SERVER['REQUEST_METHOD'] != 'POST') {
-    $inventory_request->addError('Method not allowed');
+    if (isset($_GET['action']) && $_GET['action'] == 'getConfig') {
+       /**
+        * Even if Fusion protocol is not supported for getConfig requests, they
+        * should be handled and answered with a json
+        */
+        $inventory_request->addError('Protocol not supported', 400);
```

Comment:
501 seems more appropriate to me.
```suggestion
        $inventory_request->addError('Not Implemented', 501);
```

Reply:
On my point of view, 4xx error tells the client does a bad thing and 5xx tells something goes wrong on server side. We come here if an agent is configured with "/front/inventory.php" in place of "/plugins/glpiinventory" in its server url. So this is better a client problem IMHO and this is why I would prefer a 400 here, at least for now.
Tell me if I'm missing something.

Reply:
OK. This is indeed a client issue.